WooCommerce/Classes/Authentication/Prologue/LoginPrologueViewController.swift

```diff
@@ -152,28 +152,19 @@ private extension LoginPrologueViewController {
     /// Returns the Disclaimer Attributed Text (which contains a link to the Jetpack Setup URL).
     ///
     var disclaimerAttributedText: NSAttributedString {
-        let disclaimerText = NSLocalizedString("This app requires Jetpack to connect to your Store. Read the ", comment: "Login Disclaimer Text and Jetpack config instructions")
+        let disclaimerText = NSLocalizedString("This app requires Jetpack to connect to your Store. <br /> Read the <a href=\"https://jetpack.com/support/getting-started-with-jetpack/\">configuration instructions</a>.", comment: "Login Disclaimer Text and Jetpack config instructions")
         let disclaimerAttributes: [NSAttributedString.Key: Any] = [
             .font: UIFont.font(forStyle: .caption1, weight: .thin),
             .foregroundColor: UIColor.white
         ]
 
-        let readText = NSLocalizedString("configuration instructions", comment: "Login Disclaimer Linked Text")
-        let readAttributes: [NSAttributedString.Key: Any] = [
-            .link: URL(string: WooConstants.jetpackSetupUrl) as Any,
-            .font: UIFont.font(forStyle: .caption1, weight: .thin)
-        ]
-
-        let readAttrText = NSMutableAttributedString(string: readText, attributes: readAttributes)
-
-        let endSentenceText = "."
-        let endSentenceAttrText = NSMutableAttributedString(string: endSentenceText, attributes: disclaimerAttributes)
+        guard let disclaimerAttrText = disclaimerText.htmlToAttributedString else {
```

Comment:
If you change the computed property to non-nil then that makes this catch not necessary (it didn't have a `throws` before anyway 😛). And if you change the computed property `NSAttributedString` you can update the code to something like:

```swift
let disclaimerAttrText = NSMutableAttributedString()
disclaimerAttrText.append(disclaimerText.htmlToAttributedString)
```

This is only a personal preference, really. I like having added functionality like this give immutable things - if the consumer wants to make it mutable that's cool - it just then makes its intention clear.

Reply:
`do / try / catch` statements don't propagate errors "up" a level, so `throw` isn't needed. Check this out: https://docs.swift.org/swift-book/LanguageGuide/ErrorHandling.html. 

I totally see your point about not making the computed property mutable or optional though. I didn't think of writing it that way!